applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -0,0 +1,2054 @@
+		rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+	if (rRightHandSideVector.size() != LocalSize)
+		rRightHandSideVector.resize(LocalSize, false);
+
+	noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+	noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+	if (TElementData::ElementManagesTimeIntegration) {
+		TElementData data;
+		data.Initialize(*this, rCurrentProcessInfo);
+
+		if (data.IsCut()) {
+			GeometryType::Pointer p_geom = this->pGetGeometry();
+            Matrix shape_functions_pos;
+            Matrix shape_functions_neg;
+            Matrix shape_functions_enr_pos;
+            Matrix shape_functions_enr_neg;
+            GeometryType::ShapeFunctionsGradientsType shape_derivatives_pos;
+            GeometryType::ShapeFunctionsGradientsType shape_derivatives_neg;
+            GeometryType::ShapeFunctionsGradientsType shape_derivatives_enr_pos;
+            GeometryType::ShapeFunctionsGradientsType shape_derivatives_enr_neg;
+            ComputeSplitting(
+                data,
+                shape_functions_pos,
+                shape_functions_neg,
+                shape_functions_enr_pos,
+                shape_functions_enr_neg,
+                shape_derivatives_pos,
+			    shape_derivatives_neg,
+                shape_derivatives_enr_pos,
+                shape_derivatives_enr_neg);
+
+			if (data.NumberOfDivisions == 1) {
+				//cases exist when the element is not subdivided due to the characteristics of the provided distance
+				//in this cases the element is treated as AIR or FLUID depending on the side
+				Vector gauss_weights; 
+                Matrix shape_functions; 
+                ShapeFunctionDerivativesArrayType shape_derivatives; 
+                this->CalculateGeometryData( 
+                    gauss_weights, shape_functions, shape_derivatives); 
+                const unsigned int number_of_gauss_points = gauss_weights.size(); 
+				array_1d<double,NumNodes> Ncenter;
+				for(unsigned int i=0; i<NumNodes; i++) Ncenter[i]=0.25;
+				const double dgauss = inner_prod(data.Distance, Ncenter);
+				for (unsigned int g = 0; g < number_of_gauss_points; g++) {
+					data.UpdateGeometryValues(g,gauss_weights[g], row(shape_functions, g),
+						shape_derivatives[g]);
+
+                    data.CalculateDensityAtGaussPoint();
+
+					if (dgauss > 0.0)
+						data.CalculateAirMaterialResponse();
+					else
+						this->CalculateMaterialResponse(data);				
+
+					this->AddTimeIntegratedSystem(
+						data, rLeftHandSideMatrix, rRightHandSideVector);
+				}
+			}
+
+			else {
+				MatrixType Vtot;
+				MatrixType Htot;
+				MatrixType Kee_tot;
+				VectorType rhs_ee_tot;
+				Vtot.resize(NumNodes*(Dim + 1), NumNodes, false);
+				Htot.resize(NumNodes, NumNodes*(Dim + 1), false);
+				Kee_tot.resize(NumNodes, NumNodes, false);
+				rhs_ee_tot.resize(NumNodes, false);
+
+                noalias(Vtot) = ZeroMatrix(NumNodes*(Dim + 1), NumNodes);
```

Comment:
I think this is a potential memory leak, since Vtot is uninitialized (its size may be whatever) and you equate it to a matrix of given size. Doing `MatrixType Vtot = ZeroMatrix(NumNodes*(Dim + 1), NumNodes);` is safer, since it initializes `Vtot` to the correct size.

Same goes for the other submatrices.